db/builder.cc

```diff
@@ -266,9 +266,12 @@ Status BuildTable(
       Slice last_tombstone_start_user_key{};
       for (range_del_it->SeekToFirst(); range_del_it->Valid();
            range_del_it->Next()) {
-        auto tombstone = range_del_it->Tombstone();
-        auto kv = tombstone.Serialize();
-        // TODO(yuzhangyu): handle range deletion for UDT in memtables only.
+        // When user timestamp should not be persisted, we logically strip a
+        // range tombstone's start and end key's timestamp (replace it with min
+        // timestamp) before passing them along to table builder and to update
+        // file boundaries.
+        auto tombstone = range_del_it->Tombstone(logical_strip_timestamp);
+        std::pair<InternalKey, Slice> kv = tombstone.Serialize();
```

Comment:
Yes, we can do that. This timestamp stripping logic indeed only applies to this one place where flush happens, so just putting the logic here makes sense. I was thinking that since tombstone already host two strings, I can save creating two more strings and some copying by using that two existing strings. WDYT?

Reply:
I see. I think it's fine either way.